h2o-py/tests/testdir_algos/automl/pyunit_automl_args.py

```diff
@@ -298,10 +298,55 @@ def test_frames_cannot_be_passed_as_key():
             assert "'{}' must be a valid H2OFrame".format(attr) in str(e)
 
 
+def test_no_time_limit_if_max_models_is_provided():
+    ds = import_dataset()
+    aml = H2OAutoML(project_name="py_no_time_limit", seed=1, max_models=1)
+    aml.train(y=ds['target'], training_frame=ds['train'])
+    max_runtime = aml._build_resp['build_control']['stopping_criteria']['max_runtime_secs']
+    max_models = aml._build_resp['build_control']['stopping_criteria']['max_models']
+    assert max_runtime == 0
+    assert max_models == 1
+
+
+def test_max_runtime_secs_alone():
+    ds = import_dataset()
+    aml = H2OAutoML(project_name="py_max_runtime_secs", seed=1, max_runtime_secs=7)
+    aml.train(y=ds['target'], training_frame=ds['train'])
+    max_runtime = aml._build_resp['build_control']['stopping_criteria']['max_runtime_secs']
+    max_models = aml._build_resp['build_control']['stopping_criteria']['max_models']
+    assert max_runtime == 7
+    assert max_models == 0
+
+
+def test_max_runtime_secs_can_be_set_in_combination_with_max_models():
+    ds = import_dataset()
+    aml = H2OAutoML(project_name="py_all_stopping_constraints", seed=1, max_models=1, max_runtime_secs=12)
+    aml.train(y=ds['target'], training_frame=ds['train'])
+    max_runtime = aml._build_resp['build_control']['stopping_criteria']['max_runtime_secs']
+    max_models = aml._build_resp['build_control']['stopping_criteria']['max_models']
+    assert max_runtime == 12
+    assert max_models == 1
+
+
+def test_default_max_runtime_if_no_max_models_provided():
+    ds = import_dataset()
+    aml = H2OAutoML(project_name="py_no_stopping_constraints", seed=1, verbosity='Info')
+    with pu.Timeout(5, on_timeout=lambda: aml._job.cancel()):
+        try:
+            aml.train(y=ds['target'], training_frame=ds['train'])
```

Comment:
@seb-h2o could not we just do this? 
```
aml.train(y=ds['target'], training_frame=ds['train'])
aml._job.cancel()
```
Or it is not enough time to get response?

Reply:
you can be certain that I tried... and it obviously doesn't work: `aml.train` is blocking. And I decided that Jenkins had a better use of its time than waiting for 1h.